### ISSUE TITLE
regarding model/20170512-110547

### ISSUE THREAD

**Issue by alokc83** [cjekel/tindetheus#56]
I got this directory. 
Everything is working fine. 
But whole deep learning piqued my interest. 

when I am training my models how I know if what I am training is going into these models. 
the process runs fine but modified dates on files in folder 20170512-110547 never get updated. 

I am only 2 weeks old in python and very curious about creating models and all. 
What is better than tinder to train models. so much data. 

if you can point me to the right directions, books, articles, blogpost or youtube. I would appreciate it. 

Thanks,
Alok C

**Comment by cjekel** [comment]
That model folder is a deep learning model trained to do facial classification on the MS Celeb 1M data set. The same basic abstraction, of turning data into a vector, is used in many deep learning models. Look at https://pytorch.org/tutorials/beginner/dcgan_faces_tutorial.html which is a nice tutorial.

I use that facenet model to turn an image of a face into a vector of numbers, this vector uniquely identifies an individual's face. 

So my actual model, is a scikit-learn model, which gets saved as a .pkl file. https://github.com/cjekel/tindetheus/blob/c5af11014ed8a5c0476e5550510ccb8d63f0ac36/tindetheus/machine_learning.py#L111

I would start small at first. Try different scikit-learn models other than the logistic regression I use, and see if you can build a better model. (This will have you do some digging on what defines a better model)

Here is an underrated lecture series on 'learning from data' https://www.youtube.com/watch?v=mbyG85GZ0PI&list=PLnIDYuXHkit4LcWjDe0EwlE57WiGlBs08 It was meant for an undergraduate. 

Once your comfortable with fitting different models to data, then I'd jump into deep learning. There's so many places to explore with deep learning.

If you manually review a lot of profiles, you can try deep learning on this image dataset http://philipperemy.github.io/tinder-deep-learning/ Also try a more 'modern' deep learning architecture like resnet https://pytorch.org/hub/pytorch_vision_resnet/ (There is no guarantee that it will work, and expect a lot of noise in actual Tinder profiles)

Hope this helps, and happy hacking!

**Comment by alokc83** [comment]
I much much appreciated this information. I am barely beginning in python. But after posting this issue, I dived headfirst. While trying to make it work, I understood what that model folder is for. Not saying I understand how it do it, but for sure, I know now what it does. 
Thank you so much for creating this. I have a running version now. I don't have a heavily trained model as process gets killed. amazingly I ran it while looking at temp images and said like or dislike and to my surprised 70% of the times my answer match with the system. 
Well who knew tindering could be so helpful in educate onself. 
It gave quite a purpose to my learning. not because of women of tinder. we all know how that goes. but how much data is my disposal to create whatever I wanna create. 
Well may be in 3-4 I will get close to what you have created. lol
